scylla/src/transport/authenticate_test.rs

```diff
@@ -21,3 +25,60 @@ async fn authenticate_superuser() {
 
     println!("Ok.");
 }
+
+struct CustomAuthenticator;
+
+#[async_trait]
+impl AuthenticatorSession for CustomAuthenticator {
+    async fn evaluate_challenge(
+        &mut self,
+        _token: Option<&[u8]>,
+    ) -> Result<Option<Vec<u8>>, AuthError> {
+        Err("Challenges are not expected".to_string())
+    }
+
+    async fn success(&mut self, _token: Option<&[u8]>) -> Result<(), AuthError> {
+        Ok(())
+    }
+}
+
+struct CustomAuthenticatorProvider;
+
+#[async_trait]
+impl AuthenticatorProvider for CustomAuthenticatorProvider {
+    async fn start_authentication_session(
+        &self,
+        _authenticator_name: &str,
+    ) -> Result<(Option<Vec<u8>>, Box<dyn AuthenticatorSession>), AuthError> {
+        let mut response = BytesMut::new();
+        let cred = "\0cassandra\0cassandra";
+        let cred_length = 20;
```

Comment:
There is absolutely no reason for hardcoding the length of the string literal from the previous line. You should just do `let cred_length = cred.len();` and that's it.

Perhaps you thought that Rust's `&str` is like `const char*` from C? Unlike C, Rust's strings are not null-terminated. The `&str` type is a pair `(pointer to the beginning of the string, length of the string)`, so the length of the string is immediately known and calling `.len()` is super cheap here and works even if there are null bytes inside it.